lib/build/builder.js

```diff
@@ -0,0 +1,215 @@
+  }
+
+  ready () {
+    if (!this.goconfig) {
+      return false
+    }
+    return true
+  }
+
+  lint (editor) {
+    if (!this.ready() || !editor) {
+      return []
+    }
+    const p = editor.getPath()
+    if (!p) {
+      return []
+    }
+    return Promise.resolve().then(() => {
+      const options = this.getLocatorOptions(editor)
+      return this.goconfig.locator.findTool('go', options).then((cmd) => {
+        if (!cmd) {
+          return []
+        }
+
+        const options = this.getExecutorOptions(editor)
+        const buildPromise = this.lintInstall(cmd, options)
+
+        let testPromise = Promise.resolve({ output: '', linterName: 'test' })
+        if (this.hasTests(p)) {
+          testPromise = this.lintTest(cmd, options)
+        }
+
+        return Promise.all([buildPromise, testPromise]).then((results) => this.getMessages(results, options.cwd))
+      })
+    }).catch((error) => {
+      if (error.handle) {
+        error.handle()
+      }
+      console.log(error)
+      return []
+    })
+  }
+
+  getMessages (results, cwd) {
+    let messages = []
+    for (const { output, linterName } of results) {
+      const newMessages = this.mapMessages(output, cwd, linterName)
+      for (const newMessage of newMessages) {
+        if (!messages.some((message) => this.messageEquals(newMessage, message))) {
+          messages = messages.concat(newMessage)
+        }
+      }
+    }
+    // add the "(<name>)" postfix to each message
+    for (const message of messages) {
+      message.text += ' (' + message.name + ')'
+    }
+    return messages
+  }
+
+  messageEquals (m1, m2) {
+    return m1.filePath === m2.filePath &&
+      m1.row === m2.row &&
+      m1.text === m2.text &&
+      JSON.stringify(m1.range) === JSON.stringify(m2.range)
+  }
+
+  lintInstall (cmd, options) {
+    const buildArgs = ['install', '.']
+    return this.goconfig.executor.exec(cmd, buildArgs, options).then((r) => {
+      if (r.stdout && r.stdout.trim() !== '') {
+        console.log('go install: (stdout) ' + r.stdout)
+      }
+      return { output: r.stderr.trim(), linterName: 'build' }
+    }).catch((e) => {
+      console.log(e)
+      return { output: '', linterName: 'build' }
+    })
+  }
+
+  lintTest (cmd, options) {
+    const tempdir = fs.realpathSync(temp.mkdirSync())
+    const testArgs = ['test', '-c', '-o', tempdir, '.']
+    return this.goconfig.executor.exec(cmd, testArgs, options).then((r) => {
+      if (r.stdout && r.stdout.trim() !== '') {
+        console.log('go test: (stdout) ' + r.stdout)
+      }
+      rimraf(tempdir, (e) => {
+        if (e) {
+          if (e.handle) {
+            e.handle()
+          }
+          console.log(e)
+        }
+      })
+      return { output: r.stderr.trim(), linterName: 'test' }
+    }).catch((e) => {
+      console.log(e)
+      return { output: '', linterName: 'test' }
+    })
+  }
+
+  getLocatorOptions (editor = atom.workspace.getActiveTextEditor()) {
```

Comment:
ok so I quickly searched for `getLocatorOptions` and compared them: 
* `godef`, `linter`, `get-manager` and `builder` use the same (linter has some additional safety `if`s)
* `formatter`, `information` and `tester` use nearly the same version although `formatter` has some relative path handling in `projectPath` that could affect the others... 🤔